tests/models/config_model/test_app.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for the AppConfig Pydantic model and associated logic."""
+
+import logging
+import pytest
+from pathlib import Path
+from typing import Any
+from unittest.mock import Mock, patch # ADDED Mock import
+
+from pydantic import ValidationError, BaseModel
+from docbuild.models.config_model.app import AppConfig
+from docbuild.config.app import PlaceholderResolutionError
+
+
+# --- Helper Fixtures ---
+
+# The test now imports Mock and uses the function name directly
+@pytest.fixture
+def mock_replace_placeholders(monkeypatch):
+    """Mocks the replace_placeholders utility to confirm it was called."""
+    mock = Mock(wraps=lambda d: d)  # Retains original behavior but tracks calls
+    monkeypatch.setattr('docbuild.models.config_model.app.replace_placeholders', mock)
```

Comment:
This is technically correct, but there is another syntax which passes the real object. I prefer the latter for one reason: if you rename the objects, it is detected by an editor (e.g. VSCode). If the argument for `setattr` is a string, it's not as straightforward.

Something like this (untested):

```suggestion
    monkeypatch.setattr(replace_placeholders, mock)
```